include/s3select_functions.h

```diff
@@ -814,13 +822,20 @@ struct _fn_to_timestamp : public base_function
       else if (frac_sec < 100000)
               frac_sec = frac_sec * 10;
 
-      //TODO: Include timezone hours(tz_hr) and timezone minutes(tz_mn) in date time calculation.
       new_ptime = boost::posix_time::ptime(boost::gregorian::date(yr, mo, dy),
                           boost::posix_time::time_duration(hr, mn, sc, frac_sec));
-
-      result->set_value(&new_ptime);
     #endif
 
+    int tz_hour = tz_hr;
+    if ((char)sign == '-')
+    {
+      tz_hour *= -1;
+    }
+
+    tmstmp = std::make_tuple(new_ptime, boost::posix_time::time_duration(tz_hour, tz_mn, 0), (char)tm_zone == 'Z');
```

Comment:
should validate that timezone-hour are between -12 --> +14

Reply:
Done